.circleci/config.yml

```diff
@@ -291,28 +318,64 @@ jobs:
           paths:
             - images
 
-  deploy-docker-image:
+  compare-docker-image:
     executor: ubuntu-builder
     environment:
-      DOCKER_REPO: trustlines/tlbc-testnet-next
       LOCAL_IMAGE: tlbc-testnet-next
     working_directory: ~/repo
     steps:
       - setup_remote_docker
+      - checkout
       - attach_workspace:
           at: '~'
       - run:
           name: Load docker image
           command: |
             du -hc ~/images/*
             docker load --input ~/images/$LOCAL_IMAGE.tar
-            docker image ls
       - run:
-          name: Login to dockerhub
+          name: Fetch latest pre-release
           command: |
-            echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USER" --password-stdin
+            docker pull trustlines/tlbc-testnet-next:pre-release
+      - run:
+          name: Compute image tree summaries
+          command: |
+            .circleci/show-docker-tree trustlines/tlbc-testnet-next:pre-release >/tmp/tree-pre-release
+            .circleci/show-docker-tree $LOCAL_IMAGE >/tmp/tree-local
       - run:
-          name: Upload latest
+          name: Compare image tree summaries
+          command: |
+            diff -s /tmp/tree-pre-release /tmp/tree-local || true
+
+  deploy-docker-image:
+    executor: ubuntu-builder
+    environment:
+      LOCAL_IMAGE: tlbc-testnet-next
+    working_directory: ~/repo
+    steps:
+      - run:
+          name: set DOCKER_REPO
+          command: |
+            echo ': \"${DOCKER_REPO:=trustlines/tlbc-testnet-next}\"' >> ${BASH_ENV}
+          # this allows us to set DOCKER_REPO from circleci when building in a
+          # fork. makes testing easier.
+      - upload-docker-image
+
+  deploy-docker-release-image:
+    executor: ubuntu-builder
+    environment:
+      LOCAL_IMAGE: tlbc-testnet-next
+    working_directory: ~/repo
+    steps:
+      - run:
+          name: set DOCKER_REPO
+          command: |
+            echo ': \"${DOCKER_REPO:=trustlines/tlbc-testnet}\"' >> ${BASH_ENV}
+          # this allows us to set DOCKER_REPO from circleci when building in a
+          # fork. makes testing easier.
+      - upload-docker-image
+      - run:
+          name: upload latest image
```

Comment:
should we maybe not upload to latest from `pre-release`

Reply:
I've done that on purpose. I don't think having the 'latest' tag available provides much value. It's unclear without further documentation what you get when you run the `latest` tag.

For the real release we need the latest tag, because that is what watchtower is using.

Reply:
Latest is the default tag that you pull if you don't provide a tag. I don't think you can remove old images, so that means that people might end up with really old images. 

I think we use latest, watchtower probably does not care and works with all tags.

Reply:
> Latest is the default tag that you pull if you don't provide a tag. I don't think you can remove old images, so that means that people might end up with really old images.

yes, it looks like we can't remove tags/images. We could remove the whole docker tlbc-testnet-next repository and recreate it. This is what I suggest we should do.